### DISCUSSION TITLE
Integration of Add-ons

### DISCUSSION THREAD

**Discussion by ionutfrent15** [scottyzen/woonuxt#150] [Ideas]
Hey @scottyzen ,

I hope you're doing well! I've been working with the repository and have a suggestion for enhancing the user interface by integrating add-ons from the Woo store.

**Proposal:**

Currently, the UI lacks support for displaying add-ons associated with products. We already have the possibility to specify add-ons for a product in the Woo store.
<img width="1079" alt="link" src="https://github.com/scottyzen/woonuxt/assets/50477824/735b2187-7574-4c86-9aa3-561359057d20">

**Implementation Ideas:**

Fetch Add-ons via API: We can extend the current API integration to fetch add-ons associated with products. This could involve modifying the existing GraphQL queries to include add-on information.

UI Components: Once we have the add-on data, we can create UI components to display them alongside the main product. This could include dropdowns, checkboxes, or any other suitable UI elements to allow users to select add-ons.

Interactions: Implement interactions to update the total price dynamically when add-ons are selected. We should also consider modifying the stock availability based on add-on selections.

**Benefits:**

Improved user experience: Users will have more flexibility and choice when selecting products.
Enhanced functionality: Integration of add-ons will make the UI more comprehensive and feature-rich.

**Discussion:**

I'd love to hear your thoughts on this proposal. Do you think integrating add-ons into the UI is a feasible and beneficial enhancement? If so, how do you envision the implementation process?

Looking forward to your feedback!

Best regards,
Ionut F.

**Comment by scottyzen** [comment]
Hi @ionutfrent15

About extending the API. I think [WooGrapghQL](https://woographql.com/) Pro already has the product add-on baked in. I just need to get around to making the UI. I wouldn't want to extend the API outside of @kidunot89 version. But I definitely agree having the UI being able to support this out of the box would be a huge boost in the user experience! I will look into as soon as I get a chance!

Kind regards
Scott
